src/lib/registration/ui/service_selection_dialog.rb

```diff
@@ -0,0 +1,154 @@
+      #
+      # @see #run
+      def self.run(services:, heading: nil, description: nil)
+        new(services: services, heading: heading, description: description).run
+      end
+
+      # Constructor
+      #
+      # @param services [Array<SlpServiceClass::Service] list of services to show
+      def initialize(services: [], heading: nil, description: nil)
+        super()
+
+        textdomain "registration"
+
+        @services = services
+        @heading = heading || _("Local Registration Servers")
+        @description = description || _("Select a detected registration server " \
+          "from the list\nor the default SUSE registration server.")
+      end
+
+      # Handler for the Ok button
+      #
+      # If no option was selected, a error message is shown.
+      def ok_handler
+        selected = Yast::UI.QueryWidget(Id(:services), :CurrentButton)
+        if !selected
+          Yast::Report.Error(_("No registration server selected."))
+        else
+          value = selected == "scc" ? :scc : services[selected.to_i]
+          finish_dialog(value)
+        end
+      end
+
+      # Handler for the cancel button
+      def cancel_handler
+        finish_dialog(:cancel)
+      end
+
+    protected
+
+      # Dialog's initial content
+      #
+      # @return [Yast::Term] Content
+      def dialog_content
+        MarginBox(2, 0.5,
+          VBox(
+            # popup heading (in bold)
+            Heading(heading),
+            VSpacing(0.5),
+            Label(description),
+            VSpacing(0.5),
+            RadioButtonGroup(
+              Id(:services),
+              Left(
+                HVSquash(
+                  VBox(*services_radio_buttons)
+                )
+              )
+            ),
+            VSpacing(Opt(:vstretch), 1),
+            button_box
+          ))
+      end
+
+      # Dialog options
+      #
+      # @return [Yast::Term] Dialog's options
+      def dialog_options
+        Yast::Term.new(:opt, :decorated)
+      end
+
+      # Return dialog's buttons
+      #
+      # @return [Yast::Term] Buttons' description
+      def button_box
+        ButtonBox(
+          PushButton(Id(:ok), Opt(:default), Yast::Label.OKButton),
+          PushButton(Id(:cancel), Yast::Label.CancelButton)
+        )
+      end
+
+      # Return service radio buttons
+      #
+      # Creates one radio button for each service and add a special
+      # one for SCC.
+      #
+      # @return [Yast::Term] Service radio button's description
+      def services_radio_buttons
+        widgets = services.map.with_index do |service, index|
+          Left(
+            RadioButton(
+              Id(index.to_s),
+              ::Registration::Helpers.service_description(service),
+              false
+            )
+          )
+        end
+
+        widgets.unshift(Left(RadioButton(Id("scc"),
+          # %s is the default SCC URL
+          _("SUSE Customer Center (%s)") % SUSE::Connect::YaST::DEFAULT_URL,
```

Comment:
This is a registration specific default. It depends if we want to make the dialog generic just for registration services or any other services as well.